content/docs/engine-dev/contributing.md

```diff
@@ -0,0 +1,182 @@
+{{< /tip >}}
+
+## Linking to source code
+
+When linking to source code on GitHub -- especially when linking to a block of code within a file -- a good best practice is to use permalinks. Documentation is a snapshot in time, and a permalink to a specific commit preserves this concept for linked source code.
+
+#### Creating file permalinks
+
+To obtain permalinks for a file in a GitHub repo, navigate to the **Code** tab and open the file, then choose **Copy permalink** from the ... menu:
+
+![Use the "Copy permalink" menu option](/images/engine-dev/creating-github-permalinks.png)
+
+To embed the link, use the standard Markdown link format.
+
+**Example**
+
+Link: `[SettingsInterface](https://github.com/o3de/o3de/blob/69dbcd08a56539315bfb0472984daf0f46e7a966/Code/Tools/ProjectManager/Source/SettingsInterface.h)`
+
+Output: [SettingsInterface](https://github.com/o3de/o3de/blob/69dbcd08a56539315bfb0472984daf0f46e7a966/Code/Tools/ProjectManager/Source/SettingsInterface.h)
+
+#### Creating line block permalinks
+
+To obtain permalinks for specific lines of code in a file in a GitHub repo:
+
+1. Navigate to the **Code** tab and open the file.
+1. Click on the first line number in the block.
+1. Shift + click on the last line number in the block.
+1. Choose **Copy permalink** from the ... menu to the left of the first line number:
+
+![Use the "Copy permalink" menu option](/images/engine-dev/creating-github-block-permalinks.png)
+
+Alternatively, you can just manually add the line number range to the end of the file's permalink URL, using the format: `#Lnnn-Lnnn`, where `nnn` represents a line number.
+
+**Example**
+
+Link: `[SettingsInterface::Get](https://github.com/o3de/o3de/blob/69dbcd08a56539315bfb0472984daf0f46e7a966/Code/Tools/ProjectManager/Source/SettingsInterface.h#L31-L44)`
+
+Output: [SettingsInterface::Get](https://github.com/o3de/o3de/blob/69dbcd08a56539315bfb0472984daf0f46e7a966/Code/Tools/ProjectManager/Source/SettingsInterface.h#L31-L44)
+
+#### Using simple links
+
+If you are linking to an entire file and are reasonably certain its filename will not change, it is acceptable to use a shorter, simple link instead.
+
+**Example**
+
+Link: `[EngineInfo.h](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/EngineInfo.h)`
+
+Output: [EngineInfo.h](https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/EngineInfo.h)
+
+## Formatting tools
+
+To help visualize your concepts, the following formatting tools are supported on the [o3de.org](https://o3de.org) website.
+
+### Creating flowcharts and class diagrams in Markdown {#diagramming-tools}
+
+You can create a variety of diagrams and other visualizations from within Markdown code blocks that will render using the Mermaid diagram tool. Refer to the [Mermaid documentation](https://mermaid-js.github.io/mermaid/) to learn about the various supported diagram types and their usage syntax.
```

Comment:
I recommend adding the project manager boot up sequence as a non-trivial example of what mermaid can do. That way other diagram makers can borrow and improve upon conventions introduced there

Reply:
Good idea. Will add this in a follow-up PR since this one has already been merged.